learn/glossary.md

```diff
@@ -0,0 +1,33 @@
+# JSON Schema Glossary
+
+This document collects short explanations of terminology one may encounter within the JSON Schema community.
+
+Whilst many of the entries below have precise technical definitions, preference is given to explanations of their conversational use, with additional references linked for further information.
+This page is not meant to be [normative](#normative), nor is it meant to contain fully original research or explanation.
+It is meant to aid the understanding of those less familiar with formal language used within JSON Schema, or within specifications more broadly.
+(In fact, entries below make effort to avoid terminology like "normative" itself for reasons just mentioned.)
+
+The entries on this page can be linked to via anchor links (e.g. `https://json-schema.org/learn/glossary.html#vocabulary`) when sharing a definition with others.
```

Comment:
There has to be a way to make the titles links so we don't have to say this.

Reply:
I assumed that happened automatically with our md setup (which apparently it doesn't so I'll look at that when I'm back at a computer), but it's still really easy to miss for newbies, so I think we should still leave this explicit note.

Reply:
Ok apparently the anchors are indeed autogenerated, so stuff works, but what's not autogenerated are those nice UI elements which when hovered show there's a link (and allow you to copy it).

I'll spend a few minutes trying to see if I can tell how to get those, but if I don't figure it out, perhaps it's not worth spending a ton more given this page (along with the rest of the site) is going to be redone? Opinions definitely welcome though.

Reply:
OK, the plugin I just added adds anchor links. It does it to all pages -- I of course can limit it to just this one if preferred?